File: python/day2/day2.py
with open('python/day2/input.txt') as f:
    data = f.read().split('\n')
horizontal = 0
depth = 0
aim = 0
for entry in data:
    if entry.startswith('forward'):
        horizontal += int(entry.split(' ')[1])
    if entry.startswith('up'):
        depth -= int(entry.split(' ')[1])
    if entry.startswith('down'):
        depth += int(entry.split(' ')[1])

print(horizontal, depth, horizontal * depth)

horizontal = 0
depth = 0
aim = 0

for entry in data:
    if entry.startswith('forward'):
        horizontal += int(entry.split(' ')[1])
        depth += aim*int(entry.split(' ')[1])
    if entry.startswith('up'):
        aim -= int(entry.split(' ')[1])
    if entry.startswith('down'):
        aim += int(entry.split(' ')[1])

print(horizontal, depth, horizontal * depth)

File: csharp/day2/day2.cs
namespace MyApp // Note: actual namespace depends on the project name.
{
    public class Program
    {
        public static void Main(string[] args)
        {
            string file_contents = System.IO.File.ReadAllText("input.txt");
            string[] data = file_contents.Split('\n');
            int answer1 = part1(data);
            int answer2 = part2(data);
            Console.WriteLine($"The answer for part 1 is {answer1}");
            Console.WriteLine($"The answer for part 2 is {answer2}");
        }
        public static int part1(string[] data)
        {
            int depth = 0;
            int horizontal = 0;
            foreach (string entry in data)
            {
                string[] unpacked_data = entry.Split(' ');
                string direction = unpacked_data[0];
                string distanceString = unpacked_data[1];
                int distance = Int32.Parse(distanceString);
                if (direction == "forward")
                {
                    horizontal += distance;
                }
                if (direction == "down")
                {
                    depth += distance;
                }
                if (direction == "up")
                {
                    depth -= distance;
                }
            }
            return horizontal*depth;
            
        }

        public static int part2(string[] data)
        {
            int depth = 0;
            int horizontal = 0;
            int aim = 0;
            foreach (string entry in data)
            {
                string[] unpacked_data = entry.Split(' ');
                string direction = unpacked_data[0];
                string distanceString = unpacked_data[1];
                int distance = Int32.Parse(distanceString);
                if (direction == "forward")
                {
                    horizontal += distance;
                    depth += distance*aim;
                }
                if (direction == "down")
                {
                    aim += distance;
                }
                if (direction == "up")
                {
                    aim -= distance;
                }
            }
            return horizontal*depth;
        }

    }
}
